store/rootmulti/store.go

```diff
@@ -550,13 +550,18 @@ func (rs *Store) handlePruning(version int64) error {
 }
 
 func (rs *Store) pruneStores() error {
-	pruningHeights := rs.pruningManager.GetPruningHeights()
-	rs.logger.Debug(fmt.Sprintf("pruning the following heights: %v\n", pruningHeights))
+	pruningHeights, err := rs.pruningManager.GetFlushAndResetPruningHeights()
+	if err != nil {
+		return err
+	}
 
 	if len(pruningHeights) == 0 {
+		rs.logger.Debug("pruning skipped, no heights to prune")
 		return nil
 	}
 
+	rs.logger.Debug(fmt.Sprintf("pruning the following heights: %v\n", pruningHeights))
```

Comment:
```suggestion
	rs.logger.Debug("pruning heights", "heights", pruningHeights)
```

You'll need to string-ify `pruningHeights`.